differ/storage.go

```diff
@@ -537,18 +537,24 @@ func (storage DBStorage) ReadLastNotifiedRecordForClusterList(
 		clusterIDs[idx] = string(entry.ClusterName)
 	}
 
+	selectQuery := `SELECT org_id, cluster, report, notified_at FROM ( SELECT DISTINCT ON (cluster) * FROM reported`
 	whereClause := fmt.Sprintf(` WHERE event_type_id = %v AND state = 1 AND org_id IN (%v) AND cluster IN (%v)`,
 		eventTarget, inClauseFromStringSlice(orgIDs), inClauseFromStringSlice(clusterIDs))
+	orderByClause := `ORDER BY cluster, notified_at DESC) t`
+	var err error
+	var rows *sql.Rows
+
+	if len(timeOffset) == 0 || strings.Split(timeOffset, " ")[0] == "0" {
```

Comment:
I'm not sure about this `strings.Split(timeOffset, " ")[0]`. Are we using the right type for the timeoffset? maybe it should be of type time.Time? it's strange that we have to split a string

Reply:
It's a string because it is passed as is to postgres. We can indeed use a time.Time, it would be more robust, but that's a change for another PR (and a new issue)

Reply:
Right now, the value used is expected to have the format of the PostgreSQL [INTERVAL type](https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-INPUT).

Reply:
ok, np!